### PR TITLE
fix(recovery): Fix CAB recovery bug

### PR DESCRIPTION
**Pull request by jophish**
### Description

See Slack thread [here](https://valora-app.slack.com/archives/C07KBTP7RC2/p1730734326274629).

### Related issues

- Fixes #[issue number here]

### Network scalability

If a new NetworkId and/or Network are added in the future, the changes in this PR will:

- [ ] Continue to work without code changes, OR trigger a compilation error (guaranteeing we find it when a new network is added)

### CODE REVIEW

**Review by MuckT — APPROVED** [review]

File: src/keylessBackup/SignInWithEmail.tsx [hunk 1]
```diff
@@ -176,10 +176,10 @@ function SignInWithEmail({ route }: Props) {
     }
   }
 
-  if (!address) {
+  if (!address && isSetup) {
```
Comment (kathaypacific) [inline src/keylessBackup/SignInWithEmail.tsx:179]: should this be `!isSetupInOnboarding` instead?

Comment (jophish) [reply src/keylessBackup/SignInWithEmail.tsx:179]: I think the only thing that using `!isSetupInOnboarding` would change is that we would _not_ spin if the user is setting up CAB from the settings screen but their wallet address doesn't exist, which, while it seems impossible, is probably a better experience? i can make that change if we like

Comment (jophish) [reply src/keylessBackup/SignInWithEmail.tsx]: KK, i've made the change here and in the [shefi PR](https://github.com/mobilestack-xyz/mobilestack-shefi/pull/16).

Comment (kathaypacific) [reply src/keylessBackup/SignInWithEmail.tsx]: oooh i see, i misunderstood the meaning of the name. i see that the `keylessBackupFlow` value is "restore" for the onboarding restore flow. as long as this also works for the onboarding flow for making a new wallet, then all good (naively i thought that we would still be in the `!address && isSetup` condition in that case)

**Review by kathaypacific — APPROVED** [review]
thank you!! 🚀

**Pull request merged**